packages/embed-services/src/ItemList.ts

```diff
@@ -0,0 +1,114 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2023 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import type { IAPIMethods } from '@looker/sdk-rtl'
+import { EntityService } from './EntityService'
+
+export const DEFAULT_TTL = 900 // 15 minutes
+
+export interface IItemList<T> {
+  /** Cache time to live in seconds, defaults to 15 minutes */
+  readonly timeToLive: number
+  items: T[]
+  readonly expiresAt: number
+  index(key?: keyof T): ItemList<T>
+  indexedItems: Record<string, T>
+  setExpiration(): void
+  clearIfExpired(): void
+  find(key: keyof T, value: any): T | undefined
+}
+
+export interface IEntityService<T> extends IItemList<T> {
+  get(id: string, cache?: boolean, ...options: any[]): Promise<T>
```

Comment:
based on expected usage patterns, I'm wondering if the cache toggle should just be one of the options you can pass in, since we could anticipate, e.g., wanting to specify `fields` more often than cache control.